### PR TITLE
Feature/scraping api

### PR DESCRIPTION
**Pull request by Supermalonz**
implementing services

### CODE REVIEW

File: app/api/v1/hello_world_api.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module V1
+  class HelloWorldAPI < Grape::API
```
Comment (t-kanin) [inline app/api/v1/hello_world_api.rb:4]: Remove this file if not use

File: app/api/v1/api.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    format :json
+    content_type :json, 'application/json; charset=utf-8'
+    version 'v1', using: :header, vendor: 'datawow'
+
+    rescue_from ActiveRecord::RecordNotFound do |_error|
+      error!(:not_found, 404)
+    end
+
+    helpers V1::Helpers::ResponseHelper
+    # helpers V1::Helpers::SharedSetterMethods
```
Comment (t-kanin) [inline app/api/v1/api.rb:16]: remove this

File: app/controllers/keywords_controller.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+class KeywordsController < ApplicationController
```
Comment (t-kanin) [inline app/controllers/keywords_controller.rb:1]: remove file if not use

File: app/services/scraper_service.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  attr_accessor :driver, :doc
+
+  def driver_init
+    @driver = Selenium::WebDriver.for(:chrome, capabilities: [selenium_options, selenium_capabilities_chrome])
+    @driver.get("https://www.google.com/search?hl=en&q=#{word}")
+  end
+
+  def ads_result
+    ads_array = Array.new
+    ads_listings = doc.css('div#tads').each do |ads_listing|
```
Comment (t-kanin) [inline app/services/scraper_service.rb:23]: optional: consider using `each_with_object`

https://apidock.com/rails/Enumerable/each_with_object

File: app/services/scraper_service.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        url: ads_listing.css('@href').first.text
+      }
+      ads_array << ads
+    end
+    ads_array.reject { |e| e[:header].empty? }
+  end
+
+  def nonads_result
+    nonads_array = Array.new
+    nonads_listings = doc.css('div.g').each do |nonads_listing|
```
Comment (t-kanin) [inline app/services/scraper_service.rb:35]: optional: consider using each_with_object

https://apidock.com/rails/Enumerable/each_with_object

**Review by t-kanin — APPROVED** [review]

File: app/models/keyword.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class Keyword < ApplicationRecord
+  has_many :search_results, dependent: :destroy
+
+  def self.import(file)
+    SmarterCSV.process(file.path) do |element|
+      ScraperService.perform_async(element.map { |e| e[:keyword]})
+    end
+  end
```
Comment (t-kanin) [inline app/models/keyword.rb:8]: For next PR, consider moving this part to service object.

File: app/workers/scrap_word_worker.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class ScrapWordWorker
```
Comment (jittarinkan) [inline app/workers/scrap_word_worker.rb:3]: please remove if unused

**Pull request merged**